boards/sodaq_sara_aff/src/lib.rs

```diff
@@ -0,0 +1,126 @@
+#![no_std]
+
+extern crate atsamd_hal as hal;
+
+#[cfg(feature = "rt")]
+extern crate cortex_m_rt;
+#[cfg(feature = "rt")]
+pub use cortex_m_rt::entry;
+
+use hal::prelude::*;
+pub use hal::target_device::*;
+pub use hal::*;
+
+use gpio::{Floating, Input, PfD, Port};
+
+use hal::clock::GenericClockController;
+use hal::sercom::{PadPin, UART0};
+use hal::time::Hertz;
+
+define_pins!(
+    /// Maps the pins to their arduino names and
+    /// the numbers printed on the board.
+    struct Pins,
+    target_device: target_device,
+
+    /// Arduino header digital pins
+    /// RX
+    pin d0 = b31,
```

Comment:
We prefer descriptive names (RX) over arduino names (D0) generally, or whatever name is silkscreened on the board/used in the official pinout diagram. Same goes for many other pins below.